src/Featurizers/Components/DocumentStatisticsEstimator.h

```diff
@@ -86,27 +49,28 @@ class DocumentStatisticsAnnotationData {
     // |  Public Types
     // |
     // ----------------------------------------------------------------------
-    using FrequencyMap                      = std::unordered_map<std::string, std::uint32_t>;
-    using IndexMap                          = FrequencyMap;
+    using FrequencyAndIndexMap              = std::unordered_map<std::string, FrequencyAndIndex>;
 
     // ----------------------------------------------------------------------
     // |
     // |  Public Data
     // |
     // ----------------------------------------------------------------------
-    FrequencyMap const                      TermFrequency;                  /// Words and the number of documents that it appears in
-    IndexMap const                          TermIndex;
+    FrequencyAndIndexMap const              TermFrequencyAndIndex;
     std::uint32_t const                     TotalNumDocuments;
 
     // ----------------------------------------------------------------------
     // |
     // |  Public Methods
     // |
     // ----------------------------------------------------------------------
-    DocumentStatisticsAnnotationData(FrequencyMap termFreq, IndexMap termIndex, std::uint32_t totalNumDocuments);
-    ~DocumentStatisticsAnnotationData(void) = default;
+    DocumentStatisticsAnnotationData(DocumentStatisticsAnnotationData &&other) :
+        TermFrequencyAndIndex(std::move(const_cast<FrequencyAndIndexMap &>(other.TermFrequencyAndIndex))),
+        TotalNumDocuments(std::move(other.TotalNumDocuments)) {
+    }
 
-    FEATURIZER_MOVE_CONSTRUCTOR_ONLY(DocumentStatisticsAnnotationData);
+    DocumentStatisticsAnnotationData(FrequencyAndIndexMap termFrequencyAndIndex, std::uint32_t totalNumDocuments);
```

Comment:
Does this class need a move assignment operator as well?

Reply:
it looks both AnnotationClass and This class do not need assignment operator